app/Models/Post.php

```diff
@@ -218,22 +218,36 @@ public function toBlinkPayload()
         // Blink expects quantity to be a number.
         $quantity = $this->quantity === null ? 0 : $this->quantity;
 
+        // Bypass Campaign->cause accessor method so the value isn't converted to an array
+        // which Customer.io does not support.
+        $campaign_cause = (string) optional($this->signup->campaign)->getAttributes()['cause'];
+
         // Fetch Campaign Website information via GraphQL.
         $campaignWebsite = app(GraphQL::class)->getCampaignWebsiteByCampaignId($this->campaign_id);
 
+        // The associated Action for this post.
+        $action = $this->actionModel;
+
         return [
             'id' => $this->id,
             'signup_id' => $this->signup_id,
             'quantity' => $quantity,
             'why_participated' => $this->signup->why_participated,
             'campaign_id' => (string) $this->campaign_id,
             'campaign_run_id' => (string) $this->signup->campaign_run_id,
-            'campaign_title' => $campaignWebsite['title'],
-            'campaign_slug' => $campaignWebsite['slug'],
+            'campaign_title' => (string) $campaignWebsite['title'],
+            'campaign_slug' => (string) $campaignWebsite['slug'],
+            'campaign_cause' => $campaign_cause,
             'northstar_id' => $this->northstar_id,
             'type' => $this->type,
             'action' => $this->getActionName(),
             'action_id' => $this->action_id,
+            'action_type' => (string) $action['action_type'],
+            'scholarship_entry' => (bool) $action['scholarship_entry'],
+            'civic_action' => (bool) $action['civic_action'],
+            'quiz' => (bool) $action['quiz'],
+            'online' => (bool) $action['online'],
+            'time_commitment' => (string) $action['time_commitment'],
```

Comment:
I wanted to flag that for customers we are limited to sending 30 attributes at a time. This PR puts us at 30. BUT I believe this is sent at an event, so it is instead limited at 10000 bytes rather than a specific number of attributes? Wanted to double check that we are okay here! Docs are [here](https://customer.io/docs/api/) under "Limits."

Reply:
👍 That was my understanding as well! I also did a bit of testing in a byte counter online and we were well below the `10000` limit even with some contrived long af dummy data.